### PR TITLE
feat: Add a compare_sizes tool.

### PR DESCRIPTION
**Pull request by eseidel**
I put this in the build_engine directory since I expect we'll
evnetually make this part of the build?  e.g. that during build
time (or shortly after?) we'll automatically compare the resulting
size of our build vs. Flutter's and fail the build if they
differ by more than our expected amount.

https://github.com/shorebirdtech/docs/issues/19

### CODE REVIEW

File: compare_sizes/bin/compare_sizes.dart [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    required this.shorebird,
+    required this.comparisonPath,
+    required this.artifactName,
+  });
+
+  factory Comparison.fromName({
+    required String artifactName,
+    required String flutterEngine,
+    required String shorebirdEngine,
+    required String comparisonPath,
```
Comment (bryanoltman) [inline compare_sizes/bin/compare_sizes.dart:76]: What is "comparisonPath"?

File: compare_sizes/bin/compare_sizes.dart [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    required this.flutter,
+    required this.shorebird,
+    required this.comparisonPath,
+    required this.artifactName,
+  });
+
+  factory Comparison.fromName({
+    required String artifactName,
+    required String flutterEngine,
+    required String shorebirdEngine,
```
Comment (bryanoltman) [inline compare_sizes/bin/compare_sizes.dart:75]: Are `flutterEngine` and `shorebirdEngine` git revs?

File: compare_sizes/bin/compare_sizes.dart [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    Comparison.fromName(
+      artifactName: 'android-arm64-release',
+      flutterEngine: flutterEngine,
+      shorebirdEngine: shorebirdEngine,
+      comparisonPath: 'flutter.jar',
+    ),
+  ];
+}
+
+void main(List<String> args) async {
```
Comment (bryanoltman) [inline compare_sizes/bin/compare_sizes.dart:138]: A short usage doc would be helpful here

**Comment by eseidel** [comment]
I refactored and added docs per your review.

**Pull request merged**